source/base/data_out_base.cc

```diff
@@ -167,6 +171,22 @@ namespace
       }
   }
 #endif
+
+
+  /**
+   * The header in binary format that the parallel intermediate files
+   * start with.
+   */
+  struct ParallelIntermediateHeaderType
+  {
+    std::uint64_t magic;
+    std::uint64_t version;
+    std::uint64_t compression;
+    std::uint64_t dimension;
+    std::uint64_t space_dimension;
```

Comment:
Is there a good reason to use `std::uint64_t` for these two fields? Same for the `compression` field.

Reply:
I added a note:
```
   * @note We are using std::uint64_t for all variables for simplicity below,
   * so that we don't have to worry about packing/data member alignment
   * by the compiler.
```
Is that acceptable? Otherwise I need to do something like ``__attribute__((__packed__));`` for this format to be portable. I am not sure if this covers all compilers.

Reply:
I don't think you need that. You write with `sizeof(...)` anyway IIRC. `sizeof` gives you how many bytes you need to write, padding or not.

Reply:
The problem is that the struct layout/padding is implementation defined with different compilers.
That would mean I can only be sure to be able to read the data if I am on the same system.
I want the format to be portable. For example: run the simulation on a cluster (x64, Intel Compiler) and then try to open the files on your laptop (apple arm, clang).
Does that make sense?

Reply:
Ah, now you're just grasping for straws not to have to do it ;-)

Structure layout is part of the OS ABI. It isn't compiler-specific, but you're right that it might differ from one system to another. So, however, is endianness, for example. Anyway, it's fine, let's not worry about it more than necessary.

Reply:
Curious, are there any relevant big endian systems out there? Running deal.II?
The bit packing of a struct seems a lot more relevant, I think.

I know there is a pack attribute for clang/GCC, but it is not standard and I have no idea about compiler support by other compilers. Do you know more?